cmd/saga/main.go

```diff
@@ -2,14 +2,163 @@ package main
+			return nil
+		}
+		if originalPreRunE != nil {
+			if err := originalPreRunE(cmd, args); err != nil {
+				return err
+			}
+		} else if originalPreRun != nil {
+			originalPreRun(cmd, args)
+		}
+
+		// use the provided token if set
+		if flag := invokedCmd.PersistentFlags().Lookup(tokenFlag); flag != nil && flag.Value.String() != "" {
+			return nil
+		}
+
+		// authenticate with the same host as api
+		host, err := extractHost(cmd)
+		if err != nil {
+			return err
+		}
+
+		accessToken, err := auth.Authenticate(invokedCmd.Context(), host, aipcli.IsVerbose(invokedCmd))
+		if err != nil {
+			return err
+		}
+		err = invokedCmd.PersistentFlags().Set(tokenFlag, accessToken)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if err := moduleCmd.Execute(); err != nil {
+		exitOnError(err)
+	}
+}
+
+func exitOnError(err error) {
+	_, _ = fmt.Fprintln(os.Stderr, err)
+	os.Exit(1)
+}
+
+// enforces that an api host must be set, either in address or host flag
+func extractHost(cmd *cobra.Command) (string, error) {
+	// check address flag
+	if flagAddress, err := cmd.Flags().GetString(addressFlag); err == nil && flagAddress != "" {
+		return flagAddress, nil
+	}
+
+	// check host flags
+	for host, hostAddress := range aipcli.GetConfig(cmd).Hosts {
+		if useHost, err := cmd.Flags().GetBool(host); err == nil && useHost {
+			return hostAddress, nil
+		}
+	}
+
+	// fail if not set
+	return "", fmt.Errorf("missing address or host flag")
+}
+
+func setUserCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   setUserCmdName,
+		Short: "Set user credentials in file saga/auth.json to authenticate with Saga cli",
+	}
+	cmd.SetOut(os.Stdout)
+	cmd.SetErr(os.Stderr)
+
+	cmd.Flags().String(clientIDFlag, "", "client ID to authenticate with")
+	cmd.Flags().Lookup(clientIDFlag).Annotations = map[string][]string{
+		authFlagAnnotation: {},
+	}
+	cmd.Flags().String(clientSecretFlag, "", "client secret to authenticate with")
+	cmd.Flags().Lookup(clientSecretFlag).Annotations = map[string][]string{
+		authFlagAnnotation: {},
+	}
+
+	cmd.RunE = func(cmd *cobra.Command, _ []string) error {
+		var authFile auth.SagaAuthFile
+		if !cmd.Flags().Changed(clientIDFlag) || !cmd.Flags().Changed(clientSecretFlag) {
+			return fmt.Errorf("missing client-id and/or client-secret flags")
+		}
+
+		clientID, err := cmd.Flags().GetString(clientIDFlag)
+		if err != nil {
+			return err
+		}
+		clientSecret, err := cmd.Flags().GetString(clientSecretFlag)
+		if err != nil {
+			return err
+		}
+
+		authFile = auth.SagaAuthFile{
+			ClientID:     clientID,
+			ClientSecret: clientSecret,
+		}
+		if err := auth.StoreCredentials(cmd, &authFile); err != nil {
+			return err
+		}
+		return nil
+	}
+	return cmd
+}
+
+func clearUserCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   clearUserCmdName,
+		Short: "Clears user credentials file saga/auth.json",
```

Comment:
Perhaps export a function from the `auth` package that returns the path and use that here:
```suggestion
		Short: "Clears user credentials file " + auth.ConfigFilePath(),
```